packages/core/src/components/tabs2/tabs2.tsx

```diff
@@ -163,9 +163,18 @@ export class Tabs2 extends AbstractComponent<ITabs2Props, ITabs2State> {
         }
     }
 
-    public componentDidUpdate(_prevProps: ITabs2Props, prevState: ITabs2State) {
+    public componentDidUpdate(prevProps: ITabs2Props, prevState: ITabs2State) {
         if (this.state.selectedTabId !== prevState.selectedTabId) {
             this.moveSelectionIndicator();
+        } else if (prevState.selectedTabId != null) {
+            const didChildrenChange = !Utils.arraysEqual(
```

Comment:
Here's your fix to your build woes:

```tsx
// comparing React nodes is difficult to do with simple logic, so
// shallowly compare their props as a workaround.
const didChildrenChange = !Utils.arraysEqual(
    this.getTabChildrenProps(prevProps),
    this.getTabChildrenProps(),
    Utils.shallowCompareKeys,
);
if (didChildrenChange) {
    this.moveSelectionIndicator();
}

// ...


private getTabChildrenProps(props: ITabs2Props & { children?: React.ReactNode } = this.props) {
    return this.getTabChildren(props).map(child => child.props);
}
```

There was a property called `_store` on each child that was being reset to a new object instance in each render cycle, causing `didChildrenChange` to always be `true`. This led to an infinite loop of `setState`s, hence your `Maximum call stack size exceeded` errors.